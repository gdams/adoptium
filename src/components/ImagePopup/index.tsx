import React, { useState } from 'react';

// create a modal popup for images to be viewed in full screen
const ImagePopup = (props) => {
    const [show, setShow] = useState(false);
    const handleShow = () => setShow(true);
    const handleClose = () => setShow(false);
    return (
      <>
        <img
          {...props}
          className='img-fluid pb-3'
          style={{ cursor: 'pointer' }}
          onClick={handleShow}
          alt={props.alt}
        />
        <div
          className={`modal fade ${show ? 'show' : ''}`}
          tabIndex={-1}
          role='dialog'
          aria-labelledby='exampleModalCenterTitle'
          aria-hidden='true'
          style={{ display: show ? 'block' : 'none' }}
        >
          <div className='modal-dialog modal-dialog-centered' style={{ maxWidth: '90%' }} role='document'>
            <div className='modal-content'>
              <img
                {...props}
                style={{ cursor: 'pointer' }}
                onClick={handleClose}
                alt={props.alt}
              />
            </div>
          </div>
        </div>
      </>
    );
};

export default ImagePopup;