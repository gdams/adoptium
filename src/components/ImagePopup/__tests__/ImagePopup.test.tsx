import React from 'react';
import { act, render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import ImagePopup from '../index';

describe('ImagePopup component', () => {
    it('should render correctly', () => {
        const { container } = render(
            <ImagePopup
                alt='test'
                src='test'
            />
        );

        // expect modal to have display: none
        const modal = container.querySelector('.modal');
        expect(modal).toHaveStyle('display: none');

        expect(container).toMatchSnapshot();
    });

    it('should expand image when clicked', () => {
        const { container } = render(
            <ImagePopup
                alt='test'
                src='test'
            />
        );

        // expect modal to have display: none
        const modal = container.querySelector('.modal');
        expect(modal).toHaveStyle('display: none');

        const image = container.querySelector('img');

        // open modal
        if (image) {
            act(() => {
                image.click();
            });
        }
        expect(modal).toHaveStyle('display: block');

        // locate modal image
        const modalImage = container.querySelector('.modal-content img');

        // close modal
        if (modalImage) {
            act(() => {
                // @ts-ignore
                modalImage.click();
            });
        }
        expect(modal).toHaveStyle('display: none');
    });
});