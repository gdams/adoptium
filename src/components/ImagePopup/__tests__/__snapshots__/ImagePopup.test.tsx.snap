// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImagePopup component > should render correctly 1`] = `
<div>
  <img
    alt="test"
    class="img-fluid pb-3"
    src="test"
    style="cursor: pointer;"
  />
  <div
    aria-hidden="true"
    aria-labelledby="exampleModalCenterTitle"
    class="modal fade "
    role="dialog"
    style="display: none;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
      role="document"
      style="max-width: 90%;"
    >
      <div
        class="modal-content"
      >
        <img
          alt="test"
          src="test"
          style="cursor: pointer;"
        />
      </div>
    </div>
  </div>
</div>
`;
